BlueprintUI/Sources/Layout/Stack.swift

```diff
@@ -70,14 +70,40 @@ extension StackElement {
     ///                     }
     ///                     ```
     ///
+    ///   - alignmentGuide: A closure that can be used to provide a custom alignment guide for this
+    ///                     child.
+    ///
+    ///                     This closure will be called with an `ElementDimensions` containing the
+    ///                     dimensions of this child, and should return a value in the child's
+    ///                     own coordinate space. This value represents the position along the
+    ///                     child's cross axis that should be aligned relative to the values of
+    ///                     its siblings.
+    ///
+    ///                     If not specified, the default value is provided by the stack's
+    ///                     alignment.
+    ///
+    ///                     The alignment guide is ignored by the `fill` alignment type.
+    ///
     ///   - key: A key used to disambiguate children between subsequent updates of the view hierarchy
     ///
     ///   - child: The child element to add to this stack
     ///
-    public mutating func add(growPriority: CGFloat = 1.0, shrinkPriority: CGFloat = 1.0, key: AnyHashable? = nil, child: Element) {
+    /// - Tag: StackElement.add
+    ///
+    mutating public func add(
+        growPriority: CGFloat = 1.0,
+        shrinkPriority: CGFloat = 1.0,
+        alignmentGuide: ((ElementDimensions) -> CGFloat)? = nil,
```

Comment:
Sorry, really late in the review with this, but why a closure here and not a `HorizontalAlignment` or `VerticalAlignment`? Would make us split out this method between the Column / Row, but feels like it makes using alignment in the common case much more straightforward/discoverable.

Reply:
Then you'd have to declare a new alignment ID type for each element you wanted to give a unique guide value, which could be cumbersome if you are aligning several elements.

I probably should have been clearer in the description about the 2 distinct features in this PR:
* Custom alignments, made with `HorizontalAlignment` and `VerticalAlignment` instances, are useful if you have an alignment that can be generalized across many cases (e.g. `top` is going to be `0` on all elements).
* Alignment guides are generally unique to the specific items you are aligning, like choosing the visual center on a particular image.

This API mimicks the [alignmentGuide modifier](https://developer.apple.com/documentation/swiftui/view/alignmentguide(_:computevalue:)-330fz) in SwiftUI, which lets you attach a closure to specify the guide value on individual elements.